pkg/ring/lifecycler.go

```diff
@@ -391,23 +404,41 @@ func (i *Lifecycler) ZonesCount() int {
 	return i.zonesCount
 }
 
+func (i *Lifecycler) Join() {
```

Comment:
It seems like if `autoJoinOnStartup` is `false` then the caller have to remember to call `Join()` right? 

I think it'w worth documenting this with Go style comments like 

// Join triggers XYZ, if you set autoJoinOnStartup to false, you must call this method